From: SeongJae Park <sj@kernel.org>
Date: Wed, 8 Oct 2025 21:08:54 -0700
Subject: [PATCH] ==== mm/damon: support pin-point targets removal ====

mm/damon: support pin-point targets removal

DAMON maintains the targets in a list, and allows committing only and
entire list of targets having new parameters.  Targets having same index
on the lists are treated as matching source and destination targets.  If
an existing target cannot find a matching one in the sources list, the
target is removed.  This means that there is no way to remove only a
specific monitoring target in the middle of the current targets list.

Such pin-point targets removal is really needed in some use cases,
though.  Monitoring accesses to virtual address spaces of processes that
spawned from a same ancestor is one example.  If a process of the group
is terminated, the user may want to remove the matching DAMON target as
soon as possible, to save in-kernel memory usage for the unnecessary
target data.  The user may also want to do that without turning DAMON
off or removing unnecessary targets, to keep the current monitoring
results for other active processes.

Extend DAMON kernel API and sysfs ABI to support the pin-point removal
in the following way.  For API, add a new damon_target field, namely
'obsolete'.  If the field on parameters commit source target is set, it
means the matching destination target is obsolete.  Than the parameters
commit logic removes the destination target from the existing targets
list.  For sysfs ABI, add a new file under target directory, namely
'obsolete_target'.  It is connected with the 'obsolete' field of the
commit source targets, so internally using the new API.

Bijan Tabatabai (bijan311@gmail.com) has originally reported this issue,
and participated in this solution design on a GitHub issue [1] for DAMON
user-space tool.

Changes from RFC
(https://lore.kernel.org/20251016214736.84286-1-sj@kernel.org)
- Wordsmith commit messages
- Add Reviewed-by: tags from Bijan

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_ickvp1nq | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_ickvp1nq

diff --git a/hkml_cv_bogus/hkml_cv_bogus_ickvp1nq b/hkml_cv_bogus/hkml_cv_bogus_ickvp1nq
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.47.3

